packages/react-events/src/FocusScope.js

```diff
@@ -135,17 +150,23 @@ const FocusScopeResponder = {
     context: ReactResponderContext,
     props: FocusScopeProps,
     state: FocusScopeState,
-  ) {
-    if (props.restoreFocus && state.nodeToRestore !== null) {
+  ): void {
+    if (
+      props.restoreFocus &&
```

Comment:
Perhaps we could both simplify the responsibilities of this component and make it more flexible for users if we instead had a callback API so we can leave it up to people to shift the focus themselves, in case they need custom behaviour.

```js
const onFocusIn = (event) => {
  // if there's a cancel button in the scope, move focus there instead of to the first focusable element
  if (cancelRef.current != null) {
    focusElement(cancelRef.current);
  }
};

const onFocusOut = (event) => {
  if (event.target != null) {
    // restore focus if menu open-button if it is still open
    focusElement(event.target);
  } else {
    // shift focus to menu open-button if menu auto-collapsed when modal opened
    focusElement(fallbackElement);
  }
};

<FocusScope onFocusIn={onFocusIn} onFocusOut={onFocusOut}>

Reply:
Sounds like a feature I was going to add anyway. I just added this to add a simple autofocus for modal cases.